hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/records/TxnAwareRecordsHistorian.java

```diff
@@ -157,6 +167,38 @@ public List<RecordStreamObject> getPrecedingChildRecords() {
         return precedingChildStreamObjs;
     }
 
+    @Override
+    public boolean throttleAllowsChildTransactions() {
+        @Nullable List<DeterministicThrottle.UsageSnapshot> snapshotsIfNeeded = null;
+        var isAllowed = true;
+        for (int i = 0, n = followingChildRecords.size(); i < n && isAllowed; i++) {
+            final var childRecord = followingChildRecords.get(i);
+            if (childRecord.recordBuilder().isPendingSuccess()) {
+                final var synthChild = childRecord.syntheticBody();
+                // We want to skip any synthetic contract creations or calls, since they don't
+                // make sense to throttle here:
+                //   (1) A synthetic ContractCreateInstance is already throttled by gas and,
+                //   optionally, the CryptoCreate throttle, if contracts.enforceCreationThrottle=true
+                //   (2) A synthetic ContractCall represents a read-only precompile, which
+                //   again is not relevant for the consensus transaction throttles
+                if (synthChild.hasContractCreateInstance() || synthChild.hasContractCall()) {
+                    continue;
+                }
+                if (snapshotsIfNeeded == null) {
+                    snapshotsIfNeeded = handleThrottling.takeSnapshots();
+                }
+                final var synthAccessor = synthAccessorFor(synthChild);
+                if (handleThrottling.shouldThrottleTxn(synthAccessor)) {
+                    isAllowed = false;
```

Comment:
and then `break`?  (And if there was a `break` then the loop condition would be simpler - in fact I think it could be a for-each loop.)

Reply:
Sonar complains if you put more than one `continue`/`break` inside a single `for` loop.